pt_miniscreen/core/components/list.py

```diff
@@ -59,6 +61,33 @@ def __init__(
         end_index = self.state["num_visible_rows"] + self.state["top_row_index"]
         self.rows = [self.create_child(Row) for Row in Rows[start_index:end_index]]
 
+    @property
+    def visible_scrollbar(self):
+        return self.state["visible_scrollbar"]
+
+    @visible_scrollbar.setter
+    def visible_scrollbar(self, value):
+        self.state.update({"visible_scrollbar": value})
+
+    @property
+    def distance_to_bottom(self):
+        max_top_row_index = len(self.state["Rows"]) - self.state["num_visible_rows"]
+        return max_top_row_index - self.state["top_row_index"]
+
+    @property
+    def distance_to_top(self):
+        return self.state["top_row_index"]
+
+    @property
+    def can_scroll_down(self):
+        next_top_row_index = self.state["top_row_index"] + 1
+        max_top_row_index = len(self.state["Rows"]) - self.state["num_visible_rows"]
+        return next_top_row_index <= max_top_row_index
+
+    @property
+    def can_scroll_up(self):
+        return self.state["top_row_index"] > 0
+
     @property
     def visible_rows(self):
```

Comment:
I think visible_rows needs to be updated to account for scroll distance, right now it assumes a scroll distance of 1